ingestion/tests/cli_e2e/test_cli_dbt_base.py

```diff
@@ -0,0 +1,173 @@
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Test DBT with CLI
+"""
+import os
+import re
+from abc import abstractmethod
+from contextlib import redirect_stdout
+from io import StringIO
+from pathlib import Path
+from typing import List
+from unittest import TestCase
+
+import pytest
+
+from metadata.cmd import metadata
+from metadata.config.common import load_config_file
+from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.tests.testCase import TestCase as OMTestCase
+from metadata.generated.schema.tests.testSuite import TestSuite
+from metadata.ingestion.api.sink import SinkStatus
+from metadata.ingestion.api.source import SourceStatus
+from metadata.ingestion.api.workflow import Workflow
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+
+PATH_TO_RESOURCES = os.path.dirname(os.path.realpath(__file__))
+
+
+class CliDBTBase(TestCase):
+    class TestSuite(TestCase):
+        catcher = StringIO()
+        openmetadata: OpenMetadata
+        dbt_file_path: str
+        config_file_path: str
+
+        # 1. deploy vanilla ingestion
+        @pytest.mark.order(1)
+        def test_connector_ingestion(self) -> None:
+            # run ingest with dbt tables
+            self.run_command(file_path=self.config_file_path)
+            result = self.catcher.getvalue()
+            self.catcher.truncate(0)
+            sink_status, source_status = self.retrieve_statuses(result)
+            self.assert_for_vanilla_ingestion(source_status, sink_status)
+
+        # 2. deploy dbt ingestion
+        @pytest.mark.order(2)
+        def test_dbt_ingestion(self) -> None:
+            # run the dbt ingestion
+            self.run_command(file_path=self.dbt_file_path)
+            result = self.catcher.getvalue()
+            self.catcher.truncate(0)
+            sink_status, source_status = self.retrieve_statuses(result)
+            self.assert_for_dbt_ingestion(source_status, sink_status)
+
+        # 3. run tests on dbt ingestion
+        @pytest.mark.order(3)
+        def test_entities(self) -> None:
+            for table_fqn in self.fqn_dbt_tables():
+                table: Table = self.openmetadata.get_by_name(
+                    entity=Table, fqn=table_fqn
+                )
+                data_model = table.dataModel
+                self.assertTrue(len(data_model.columns) > 0)
+                self.assertIsNotNone(data_model.rawSql)
+                self.assertIsNotNone(data_model.sql)
+                self.assertIsNotNone(data_model.upstream)
+                self.assertIsNotNone(data_model.description)
+                self.assertTrue(len(data_model.tags) > 0)
```

Comment:
tags are validated here

Reply:
cool just saw your message below. Thank you